Controller/CRUDController.php

```diff
@@ -511,7 +511,15 @@ protected function redirectTo($object, Request $request = null)
         }
 
         if (!$url) {
-            $url = $this->admin->generateObjectUrl('edit', $object);
+            foreach (array('edit', 'show') as $route) {
+                if ($this->admin->hasRoute($route) && $this->admin->isGranted(strtoupper($route))) {
+                    return new RedirectResponse($this->admin->generateObjectUrl($route, $object));
```

Comment:
After reflection, all case set a `$url` instead returning directly a response.

Please go back on break method to keep same logic.

Sorry for that. :-')